### ISSUE TITLE
label-has-for bug

### ISSUE THREAD

**Issue by badtant** [evcohen/eslint-plugin-jsx-a11y#282]
The following code gives me an error for jsx-a11y/label-has-for

```
<label htmlFor="test">test</label>
<input type="text" id="test">
```

**Comment by ljharb** [comment]
The default is that both nesting and an ID must be used to link the input and label:
```jsx
<label htmlFor="test">
  test
  <input type="text" id="test" />
</label>
```

**Comment by badtant** [comment]
ahh, now I see. shouldn't the default be "some" instead of "every"?

**Comment by ljharb** [comment]
@badtant nope, the default should be "every" - both are needed because some, say, 95% of assistive devices support nesting, and an overlapping 95% support ID-linking - but the only way to hit 100% is by implementing both.

It's possible that all the assistive devices that lack support for nesting are no longer in circulation, but since nobody seems to have taken the time to do the research to verify it, the proper default is to err on the side of caution.
